lib/core/dtype.cpp

```diff
@@ -16,6 +16,8 @@ bool is_fundamental(DType tp) {
   return is_int(tp) || is_float(tp) || tp == dtype<bool>;
 }
 
+bool is_total_orderable(DType tp) { return is_fundamental(tp); }
```

Comment:
What about `time_point` and `std::string`?

Reply:
Right, `time_point` should be supported. But `string` does not work. And it did not work before in label-based slicing.